pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1300,3 +1299,23 @@ func getProviderID(machine *v1alpha1.Machine) string {
 func getNodeName(machine *v1alpha1.Machine) string {
 	return machine.Status.Node
 }
+
+func (c *controller) patchNodeWithAnnotation(node *v1.Node, annotationKey string) {
+
+	// Initialize node annotations if empty
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+
+	//append annotation
+	node.Annotations[annotationKey] = "1"
```

Comment:
because of this logic, this method may not be reused elsewhere is what I feel. I would like to get advice from other reviewers as well.